### ISSUE TITLE
[BUG] Iris test dataset label is infered as regression

### ISSUE THREAD

**Issue by matanper** [deepchecks/deepchecks#1274]
```
import pandas as pd
from deepchecks.tabular import Dataset
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split

iris = load_iris(as_frame=True)
frame = iris.frame
X = iris.data
y = iris.target
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=55)
train = Dataset(pd.concat([X_train, y_train], axis=1), features=iris.feature_names, label='target')
test = Dataset(pd.concat([X_test, y_test], axis=1), features=iris.feature_names, label='target')

print(train.label_type) -> 'classification_label'
print(test.label_type) -> 'regression_label'
```

**Comment by JKL98ISR** [comment]
could be a bit hard to easily solve as we the function that infers it does according to many parameters that we might need to tweek.

**Comment by noamzbr** [comment]
Things that can be done:

1. is_float_dtype - add logic that looks for actual floats (e.g. mod 1). 
2. context - if model is present, can use it to override the dataset task type.
3. Consider - removing / updating the min_cateogrical_ratio logic.
